File: Archive/Baekjoon/~23.12.31/#08 기본 수학 2/4948_베르트랑 공준 (22.04.08 성공).cpp
#include <iostream>
#include <algorithm>

using namespace std;

bool arr[246913];

void isPrime() {
    int i = 2;
    arr[0] = false;
    arr[1] = false;

    while (i <= 246912) {
        if (arr[i])
            for (int j = i + i; j <= 246912; j += i)
                if (arr[j])
                    arr[j] = false;
        i++;
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    int N;
    cin >> N;

    fill(arr, arr + 246913, true);
    isPrime();

    while (N) {
        int res = 0;

        for (int i = N + 1; i <= 2 * N; ++i)
            if (arr[i])
                res++;

        cout << res << "\n";

        cin >> N;
    }
    
    return 0;
}

File: Archive/Baekjoon/~23.12.31/#08 기본 수학 2/4948_베르트랑 공준 (22.04.08 성공).py
arr = [False, False] + [True] * (246912 - 1)
i = 2

# 에라토스테네스의 체
while i <= 246912:
    if arr[i]: # 지워나가는 과정
        for j in range(i + i, 246912 + 1, i):
            if arr[j]:
                arr[j] = False
    i += 1

while True:
    N = int(input())
    if (N == 0):
        break
    res = 0

    for i in range(N + 1, 2 * N + 1):
        if (arr[i]):
            res += 1
    
    print(res)